schemas/com.mbta.rtr.light-rail-vehicle-information.json

```diff
@@ -0,0 +1,100 @@
+  "$id": "https://github.com/mbta/schemas/blob/main/schemas/com.mbta.rtr.light-rail-vehicle-information.json",
+  "description": "Light Rail Vehicle Information",
+  "type": "object",
+  "properties": {
+    "id": {
+      "$ref": "cloudevents.json#/properties/id",
+      "examples": ["123abc"]
+    },
+    "source": {
+      "$ref": "cloudevents.json#/properties/source",
+      "examples": ["mbta.com/rtr-1"]
+    },
+    "specversion": {
+      "const": "1.0",
+      "$ref": "cloudevents.json#/properties/specversion"
+    },
+    "type": {
+      "const": "com.mbta.rtr.light-rail-vehicle-information",
+      "$ref": "cloudevents.json#/properties/type"
+    },
+    "time": {
+      "$ref": "cloudevents.json#/properties/time"
+    },
+    "data": {
+      "description": "The event payload.",
+      "$ref": "#/$defs/datadef"
+    }
+  },
+  "required": ["type", "specversion", "source", "id", "time", "data"],
+  "$defs": {
+    "datadef": {
+      "type": "object",
+      "required": [
+        "directionId",
+        "lastAvi",
+        "vehicleId",
+        "routeId",
+        "consist",
+        "orientation",
+        "nextStopId",
+        "stopStatus",
+        "lat",
+        "lon"
+      ],
+      "properties": {
+        "directionId": {
+          "description": "Direction in which the vehicle is headed. Derived from the direction in trips.txt: https://gtfs.org/documentation/schedule/reference/#tripstxt",
+          "type": "integer"
+        },
+        "lastAvi": {
+          "description": "The last AVI code that was emitted from the vehicle. If you are internal to the MBTA, see the documentation on AVI codes here https://github.com/mbta/rtr/blob/f734e361cb7361ec0cde461add8e1e1bfc938b74/docs/avi_codes.pdf",
+          "type": "string",
+          "examples": ["875"]
+        },
+        "vehicleId": {
+          "description": "Unique identifier of the vehicle from OCS",
+          "type": "string",
+          "examples": ["G-10023"]
+        },
+        "routeId": {
+          "description": "Route that RTR thinks the vehicle is serving. Value is from GTFS trips.txt: https://gtfs.org/documentation/schedule/reference/#tripstxt",
+          "type": "string",
+          "examples": ["Green-D", "Green-C", "Mattapan"]
+        },
+        "consist": {
+          "description": "Array containing car numbers of the train in order",
+          "type": "string",
+          "examples": [["3644", "3862"]]
+        },
+        "orientation": {
+          "description": "Whether the 'rear' or 'front' of each train is facing forward for each train in the consist. B means backward, a means forward",
+          "type": "string",
+          "examples": ["AB", "BA", "BB"]
+        },
+        "nextStopId": {
+          "description": "Identifier for the next stop that the vehicle is at / approaching. From GTFS stops.txt: https://gtfs.org/documentation/schedule/reference/#stopstxt",
+          "type": "string",
```

Comment:
For all the fields in this file: Are any of them nullable? I like non-nullable fields, but if any are supposed to be nullable, in JSON Schema you'd have to specifically say they are with `"type": ["other type e.g. string", "null"]`.

For all the string fields: A common bug in producers is that a string is empty when it shouldn't be. If this schema is ever used for validation, you could hedge against that by putting `"minLength": 1` on any string fields that shouldn't be empty.

Reply:
yes, our [vehicle struct](https://github.com/mbta/rtr/blob/86824e5f326ed0625d00245095d510b5b372ddad/lib/rtr/vehicles/vehicle.ex#L53-L72) has almost every field as nullable for better or worse 😓 

would you like us to only send you vehicles in which all of these fields are populated? we could always filter on the RTR side